### PR TITLE
kv/client: fix region worker exited error

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Fix `region_worker` does not exit when TiKV store disconnect happens and region worker is set to exit manually. Unit test will be added later. Fix https://github.com/pingcap/ticdc/issues/1905 and #1904

### What is changed and how it works?

The `eventHandler` should return a non-nil error when region worker is notified to exit, which will terminate other routines under the same `errgroup` to exit

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- No release note

### CODE REVIEW

**Review by leoppro — APPROVED** [review]

File: cdc/kv/region_worker.go [hunk 1]
```diff
@@ -471,9 +474,9 @@ func (w *regionWorker) eventHandler(ctx context.Context) error {
 		if err != nil {
 			return err
 		}
-		exitEventHandler, skipEvent, err := preprocess(event, ok)
+		exitEventHandler, skipEvent, _ := preprocess(event, ok)
```
[the remark this answers is not in the document]
Comment (amyangfei) [reply cdc/kv/region_worker.go]: The error is logged in `preprocess`, and we don't need this error here. So I remove the returned error from `preprocess`

**Review by liuzix — APPROVED** [review]

**Pull request merged**